File: app/src/main/java/com/hlabexamples/fabrevealmenu/MainActivity.java
package com.hlabexamples.fabrevealmenu;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.android.material.navigation.NavigationBarView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.view.MenuItem;

public class MainActivity extends AppCompatActivity implements NavigationBarView.OnItemSelectedListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        BottomNavigationView navigation = findViewById(R.id.navigation);
        navigation.setOnItemSelectedListener(this);

        showXmlFragment();

    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        // Handle navigation view item clicks here.
        int id = item.getItemId();

        if (id == R.id.nav_xml) {
            showXmlFragment();
            return true;
        } else if (id == R.id.nav_code) {
            showCodeFragment();
            return true;
        } else if (id == R.id.nav_custom) {
            Intent intent = new Intent(MainActivity.this, ScrollingActivity.class);
            startActivity(intent);
            return true;
        }
        return false;
    }

    private void showXmlFragment() {
        getSupportFragmentManager().beginTransaction().replace(R.id.container, new DemoXmlFragment()).commit();
    }

    private void showCodeFragment() {
        getSupportFragmentManager().beginTransaction().replace(R.id.container, new DemoCodeFragment()).commit();
    }

}

File: app/src/main/java/com/hlabexamples/fabrevealmenu/MainActivity.kt
package com.hlabexamples.fabrevealmenu

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemSelectedListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        val navigation = findViewById<BottomNavigationView>(R.id.navigation)
        navigation.setOnNavigationItemSelectedListener(this)

        showXmlFragment()

    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        // Handle navigation view item clicks here.
        val id = item.itemId

        when (id) {
            R.id.nav_xml -> {
                showXmlFragment()
                return true
            }
            R.id.nav_code -> {
                showCodeFragment()
                return true
            }
            R.id.nav_custom -> {
                val intent = Intent(this@MainActivity, ScrollingActivity::class.java)
                startActivity(intent)
                return true
            }
            else -> return false
        }
    }

    private fun showXmlFragment() {
        supportFragmentManager.beginTransaction().replace(R.id.container, DemoXmlFragment()).commit()
    }

    private fun showCodeFragment() {
        supportFragmentManager.beginTransaction().replace(R.id.container, DemoCodeFragment()).commit()
    }

}
